yaci-indexer/src/main/java/org/cardanofoundation/rosetta/yaciindexer/stores/txsize/CustomTransactionSizeStore.java

```diff
@@ -0,0 +1,229 @@
+
+@Component
+@RequiredArgsConstructor
+@Slf4j
+public class CustomTransactionSizeStore {
+
+  private final TransactionSizeRepository transactionSizeRepository;
+
+  @EventListener
+  @Transactional
+  public void handleTransactionEvent(TransactionEvent transactionEvent) {
+    List<TransactionSizeEntity> transactionSizeEntities = new ArrayList<>();
+    transactionEvent.getTransactions().forEach(tx -> {
+
+      // Reconstructing Transaction based on CDDL specification - https://github.com/IntersectMBO/cardano-ledger/blob/master/eras/alonzo/impl/cddl-files/alonzo.cddl
+      Map signedTransaction = new Map();
+
+      DataItem txBody = CborSerializationUtil.deserialize(HexUtil.decodeHexString(tx.getBody().getCbor()))[0];
+      signedTransaction.put(new UnsignedInteger(TransactionBuildingConstants.TX_BODY_INDEX), txBody);
+      int scriptSize = addWitnessSetToSignedTransaction(tx, signedTransaction);
+
+      if(tx.getBlockNumber() > TransactionBuildingConstants.ALONZO_START_BLOCKNUMBER) { // starting from alonzo era
+        signedTransaction.put(new UnsignedInteger(
+            TransactionBuildingConstants.SUCCESS_INDICATOR_INDEX), SimpleValue.TRUE);
+      }
+
+      addAuxDataToTransaction(tx, signedTransaction);
+
+
+      byte[] serialize = CborSerializationUtil.serialize(signedTransaction);
+
+
+      int txSize = serialize.length;
+      // Save the transaction
+      TransactionSizeEntity transactionSizeEntity = TransactionSizeEntity.builder()
+          .txHash(tx.getTxHash())
+          .size(txSize)
+          .scriptSize(scriptSize)
+          .blockNumber(tx.getBlockNumber())
+          .build();
+      transactionSizeEntities.add(transactionSizeEntity);
+    });
+
+    if(!transactionSizeEntities.isEmpty()) {
+      transactionSizeRepository.saveAll(transactionSizeEntities);
+    }
+  }
+
+  /**
+   * Reconstructing the Witness_set from transaction and adding it to signedTransaction
+   * CDDL Definition: https://github.com/IntersectMBO/cardano-ledger/blob/e6b6d4f85fb72b5cb5b5361e534d3bb71bb9e55e/eras/alonzo/impl/cddl-files/alonzo.cddl#L269
+   * @param tx
+   * @param signedTransaction
+   * @return
+   */
+  private static int addWitnessSetToSignedTransaction(Transaction tx, Map signedTransaction) {
+    // adding witnesses to signedTransaction
+    Map witnessSet = new Map();
+    int scriptSize = 0;
+    addvKeyWitnessToWitness(tx, witnessSet);
+    addNativeScriptsToWitness(tx, witnessSet);
+    addBootstrapToWitness(tx, witnessSet);
+
+    scriptSize += addPlutusToWitness(witnessSet, tx.getWitnesses().getPlutusV1Scripts(),
+        TransactionBuildingConstants.PLUTUSV1_WITNESS_INDEX);
+    scriptSize += addPlutusToWitness(witnessSet, tx.getWitnesses().getPlutusV2Scripts(), TransactionBuildingConstants.PLUTUSV2_WITNESS_INDEX);
+    scriptSize += addPlutusToWitness(witnessSet, tx.getWitnesses().getPlutusV3Scripts(), TransactionBuildingConstants.PLUTUSV3_WITNESS_INDEX);
+
```

Comment:
After adding Plutus v4, the code should be changed here too. Do we have any storage of the available scrip versions? Maybe we could implement it later.